AK/String.cpp

```diff
@@ -124,31 +112,25 @@ void StringData::compute_hash() const
 
 String String::from_utf8_without_validation(ReadonlyBytes bytes)
 {
-    if (bytes.size() <= MAX_SHORT_STRING_BYTE_COUNT) {
-        ShortString short_string;
-        if (!bytes.is_empty())
-            memcpy(short_string.storage, bytes.data(), bytes.size());
-        short_string.byte_count_and_short_string_flag = (bytes.size() << 1) | SHORT_STRING_FLAG;
-        return String { short_string };
-    }
-    auto data = MUST(Detail::StringData::from_utf8(reinterpret_cast<char const*>(bytes.data()), bytes.size()));
-    return String { move(data) };
+    String result;
+    MUST(result.replace_with_new_string(bytes.size(), [&](u8* buffer) {
+        memcpy(buffer, bytes.data(), bytes.size());
```

Comment:
I'm *certain* Span has an overload for copying data to a buffer! (yes you need to construct an ad-hoc span)